src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs

```diff
@@ -90,17 +94,26 @@ private MergedModuleDeclaration BuildMergedRoot()
                     path, this.RootPath));
 
                 // Add to root so the compilation can continue
-                modules.Add(new SingleModuleDeclaration(rootName, rootNameSplitPath, (CompilationUnitSyntax)tree.Root));
+                modules.Add(new SingleModuleDeclaration(
+                    name: rootPath.Last,
+                    path: rootPath.Slice(^1..),
+                    syntax: (CompilationUnitSyntax)tree.Root));
                 continue;
             }
 
-            var subPath = path.RemovePrefix(this.compilation.SplitRootModulePath, true);
-            if (subPath.IsEmpty) subPath = rootNameSplitPath;
-            var name = subPath.Parts.Span[^1];
+            var subPath = path.RemovePrefix(pathBeforeRoot);
```

Comment:
does this work? don't we need to have the last element or root here too?

Reply:
So my idea was that we strip off everything _before_ the root. Then every path from the root module works with paths relative to root